Source/WebKit/WebProcess/GPU/GPUProcessConnection.cpp

```diff
@@ -103,35 +102,15 @@
 namespace WebKit {
 using namespace WebCore;
 
-static GPUProcessConnectionParameters getGPUProcessConnectionParameters()
+Ref<GPUProcessConnection> GPUProcessConnection::create(Ref<IPC::Connection> connection)
 {
-    GPUProcessConnectionParameters parameters;
-#if PLATFORM(COCOA)
-    parameters.webProcessIdentity = ProcessIdentity { ProcessIdentity::CurrentProcess };
-#endif
-    return parameters;
-}
-
-RefPtr<GPUProcessConnection> GPUProcessConnection::create(IPC::Connection& parentConnection)
-{
-    auto connectionIdentifiers = IPC::Connection::createConnectionIdentifierPair();
-    if (!connectionIdentifiers)
-        return nullptr;
-
-    RELEASE_ASSERT_WITH_MESSAGE(WebProcess::singleton().hasEverHadAnyWebPages(), "GPUProcess preferences come from the pages");
-    parentConnection.send(Messages::WebProcessProxy::CreateGPUProcessConnection(WTFMove(connectionIdentifiers->client), getGPUProcessConnectionParameters()), 0, IPC::SendOption::DispatchMessageEvenWhenWaitingForSyncReply);
-
-    auto instance = adoptRef(*new GPUProcessConnection(WTFMove(connectionIdentifiers->server)));
-#if ENABLE(IPC_TESTING_API)
-    if (parentConnection.ignoreInvalidMessageForTesting())
-        instance->connection().setIgnoreInvalidMessageForTesting();
-#endif
+    auto instance = adoptRef(*new GPUProcessConnection(WTFMove(connection)));
     RELEASE_LOG(Process, "GPUProcessConnection::create - %p", instance.ptr());
     return instance;
 }
 
-GPUProcessConnection::GPUProcessConnection(IPC::Connection::Identifier&& connectionIdentifier)
-    : m_connection(IPC::Connection::createServerConnection(connectionIdentifier))
+GPUProcessConnection::GPUProcessConnection(Ref<IPC::Connection> connection)
```

Comment:
`Ref<IPC::Connection>&&` or `IPC::Connection&` but we shouldn't be passing smart pointers by value.

Reply:
Passing Ref<T> is more correct and optimal than passing Ref<T>&&.

More correct: 

communicates and forces ownership transfer (of the outstanding reference, in this case). The caller won't have any references to the instance during the call.

More optimal:

The instance may be created into the call frame
As opposed to:
The instance is created in the caller frame and then redundant pointer passed in the call frame.

https://stackoverflow.com/questions/71930850/why-not-to-accept-stdunique-ptr-by-rvalue-reference

Reply:
This is not the pattern we use in WebKit. The reason being that this doesn't force the caller to WTFMove() and thus is likely to cause unnecessary ref-counting churn in practice.